vm/src/dictdatatype.rs

```diff
@@ -275,6 +284,20 @@ impl<T: Clone> Dict<T> {
         Ok(ret)
     }
 
+    pub fn get2<K: DictKey>(
```

Comment:
I renamed it get_chain; I could make it `pub(crate)` if you want, but I think it's best to keep the dict implementation in the dict implementation- `builtins::dict` already has all the logic regarding exact instance vs subclass checking, so I think it makes sense to keep it there rather than move it to `frame`.